components/kyma-operator/pkg/kymainstallation/step.go

```diff
@@ -123,7 +123,29 @@ func (s upgradeStep) Run() error {
 		releaseOverrides)
 
 	if upgradeErr != nil {
-		return errors.New("Helm upgrade error: " + upgradeErr.Error())
+		upgradeErrMsg := fmt.Sprintf("Helm install error: %s", upgradeErr.Error())
+		errorMsg := upgradeErrMsg
+
+		if s.deployedRevision > 0 {
```

Comment:
This condition is always satisfied. If deployedRevision == 0, it's an install step.